contracts/MerkleProof.sol

```diff
@@ -0,0 +1,41 @@
+pragma solidity ^0.4.11;
+
+/*
+ * @title MerkleProof
+ * @dev Merkle proof verification
+ * @note Based on https://github.com/ameensol/merkle-tree-solidity/blob/master/src/MerkleProof.sol
+ */
+library MerkleProof {
+  /*
+   * @dev Verifies a Merkle proof proving the existence of a leaf in a Merkle tree. Assumes that each pair of leaves
+   * and each pair of pre-images is sorted.
+   * @param _proof Merkle proof containing sibling hashes on the branch from the leaf to the root of the Merkle tree
+   * @param _root Merkle root
+   * @param _leaf Leaf of Merkle tree
+   */
+  function verifyProof(bytes _proof, bytes32 _root, bytes32 _leaf) constant returns (bool) {
+    // Check if proof length is a multiple of 32
+    if (_proof.length % 32 != 0) return false;
+
+    bytes32 proofElement;
+    bytes32 computedHash = _leaf;
```

Comment:
I think it makes sense for `_leaf` to be the hash of the leaf element instead of the actual leaf element since it avoids an additional `keccak256` operation on-chain - the more logic that can be pushed off-chain the better imo. Then, the comments/documentation can make it clearer that `_leaf` should be the hash of the leaf element and not the actual leaf element.